src/BCMathExtended/BC.php

```diff
@@ -332,44 +248,36 @@ public static function sub($leftOperand, $rightOperand, $scale = null)
         return bcsub($leftOperand, $rightOperand, $scale);
     }
 
-    /**
-     * @param $number
-     * @return bool
-     */
-    private static function isNegative($number)
+    protected static function trimTrailingZeroes(string $number): string
+    {
+        if (false !== strpos($number, '.')) {
+            $number = rtrim($number, '0');
+        }
+
+        return rtrim($number, '.') ?: '0';
+    }
+
+    protected static function isNegative(string $number): bool
     {
         return 0 === strncmp('-', $number, 1);
     }
 
-    /**
-     * @param int|string $min
-     * @param int|string $max
-     * @return string
-     */
-    public static function rand($min, $max)
+    public static function rand(string $min, string $max): string
     {
         $max = self::convertScientificNotationToString($max);
         $min = self::convertScientificNotationToString($min);
 
-        $difference = self::add(self::sub($max, $min), 1);
-        $randPercent = self::div(mt_rand(), mt_getrandmax(), 8);
+        $difference = self::add(self::sub($max, $min), '1');
+        $randPercent = self::div((string)mt_rand(), (string)mt_getrandmax(), 8);
 
         return self::add($min, self::mul($difference, $randPercent, 8), 0);
     }
 
-    /**
-     * @param array|int|string,...
-     * @return null|string
-     */
-    public static function max()
+    public static function max(/* args... */): ?string
```

Comment:
The possible type hint is missed?

Reply:
as I use func_get_args I don't see any correct phpdoc  to use here maybe 

`mixed $args,... ?`

Reply:
@krowinski, you can consider about following picture about the multiple args :).

![image](https://user-images.githubusercontent.com/9021747/56607343-e9e8df80-663a-11e9-8347-f25fbab604a5.png)

Reply:
cool splat will work here ty

Reply:
At least with doc block please. I am sure there is a way, because many static code analyzers understand more complex typehints in the doc block.

Reply:
Most basic is

@param string ...$args

which at least phan understands.

Reply:
```php
/**
 * @param string ...$args
 */
public static function max(...$args): ?string
```

Reply:
Ok I see it got already fixed with phpdoc style. nevermind then.

Reply:
`@param string $args,...` should also work.